### PR TITLE
Flipbook Viewer: Display this viewer for non-transcription multiple-image-subjects

### PR DESCRIPTION
**Pull request by goplayoutside3**
## Package
lib-classifier

## Linked Issue and/or Talk Post
Flipbook Project Board: https://github.com/zooniverse/front-end-monorepo/projects/21

## Describe your changes
ADR 46 describes the intended flipbook features and that this viewer is completely separate from the Multiframe Viewer.

Development of the Flipbook Viewer is divided into bite-size PR's. This PR sets up the basic classifier configuration and code files to display the flipbook for subjects with multiple images in the `locations` array. The flipbook is now the default viewer for that subject type unless a workflow config specifies `multiFrame`. See SubjectStore > Subject model > getViewer() for more details.

I included a basic storybook file and placeholder unit test file. The Flipbook Viewer component itself has a simple placeholder Grommet Box for now.

## How to Review

To see the Flipbook Viewer run locally, visit the following projects. 
- [Wildwatch Burrowing Owl](https://local.zooniverse.org:3000/projects/sandiegozooglobal/wildwatch-burrowing-owl/classify/workflow/21141?env=production)
- [A Flipbook Test Project](https://local.zooniverse.org:3000/projects/goplayoutside3/fem-flipbook-viewer/classify/workflow/22532?env=production)

Live FEM transcription projects should still show the Multiframe Viewer even though their subject type matches the flipbook. Examples: [Poets & Lovers](https://local.zooniverse.org:3000/projects/pmlogan/poets-and-lovers?env=production) or [Davy Notebooks](https://local.zooniverse.org:3000/projects/humphrydavy/davy-notebooks-project?env=production) because their workflow config includes `multiFrame` via their admin pages.

You can also see the placeholder Flipbook component in the classifier's storybook.

# Checklist

## General
- [x] Tests are passing locally and on Github
- [x] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [x] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [x] Can submit a classification
- [x] Can sign-in and sign-out

## New Feature
- [x] The PR creator has listed user actions to use when testing the new feature
- [x] Unit tests are included for the new feature
- [x] A storybook story has been created or updated
  - Example of SlideTutorial [component](https://github.com/zooniverse/front-end-monorepo/blob/master/packages/lib-classifier/src/components/Classifier/components/SlideTutorial/SlideTutorial.js) with docgen comments, and its [story](https://zooniverse.github.io/front-end-monorepo/@zooniverse/classifier/index.html?path=/docs/other-slidetutorial--default)

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Will there be merge conflicts here with #3694, which refactors the `Subject` models in the classifier store?

**Comment by goplayoutside3** [comment]
> Will there be merge conflicts here with #3694, which refactors the `Subject` models in the classifier store?

It's possible, but the lines I added to Subject.js are minimal and building Flipbook is prioritized before PR's labeled for maintenance week.

**Comment by eatyourgreens** [comment]
True. That PR adds explicit typing and tests for subjects with multiple images, so I wanted to double check that doesn’t clash with the tests here.

**Review by eatyourgreens — APPROVED** [review]

**Comment by goplayoutside3** [comment]
Thanks for the approval @eatyourgreens. I think Mark is assigned to review this PR too, but in the future please use the review checklist for my authored PR's - I'd like the confidence that someone has checked my code changes run as expected in the classifier.

**Comment by eatyourgreens** [comment]
It ran fine for me in the classifier this morning.

File: docs/arch/adr-46.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+The Flipbook Viewer will have the same features as PFE, and its design is catered to landscape oriented subject images. [Invision Design](https://projects.invisionapp.com/prototype/Flipbook-Viewer-cl7ar8qa402m83i01w6uvj5hq)
+
+## Features
+
+Features of the FEM Flipbook Viewer will include:
+1. Thumbnails of each subject's images below the currently selected image. Clicking these buttons will change the currently selected image. 
+2. Next/Back buttons to navigate through available thumbnails.
+3. Play/Pause to flip/animate through the images.
+4. Switching between the "flipbook" display and "separate images" display.
```
Comment (mcbouslog) [inline docs/arch/adr-46.md:21]: I think it is, but confirming that the initial plan is to only implement the separate frame viewer with images in a single column? And the separate frame viewer with 2 or 3 columns will be implemented later. If so, maybe add one sentence accordingly.

**Review by mcbouslog — APPROVED** [review]
Tested on all the noted workflows (flipbook, multiframe, single image) and image and text and everything worked as expected (not flipbook workflows looked the same 👍 ).

Something to consider, though I could be missing big picture pieces - 
In PFE a workflow can be configured to only be separate frames, like [this workflow](https://master.pfe-preview.zooniverse.org/projects/markb-panoptes/test-project-1-mb-fem-lab/classify?reload=0&workflow=3429&env=staging) (not toggle-able to flipbook), but for FEM are we treating separate frames as a feature on the flipbook viewer? Maybe the separate frame viewer should be its own viewer, and the related flipbook/separate frame toggle button toggles between the two different viewers? So a multi image subject would by default be flipbook, or could be set to multi-frame or seperate-frame viewers. I'm not sure if any live/launched projects are exclusively separate frames though.

**Comment by goplayoutside3** [comment]
> for FEM are we treating separate frames as a feature on the flipbook viewer? 

@mcbouslog do you know if PFE builds them as completely separate viewers? Or is separate frames a feature of flipbook?

**Comment by mcbouslog** [comment]
@goplayoutside3 - I'm looking through the [PFE SubjectViewer file](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/components/subject-viewer.cjsx#L134-L146) and I think it depends on how the few lines linked are interpreted. It's analyzing the subject and returning different variations on an image subject viewer (image and audio, flipbook, grouped subject, or separate frames). Are each of those variations on an image subject viewer in PFE different subject viewers in FEM? I think so, but could probably be easily convinced otherwise.

I don't think this should hold up this PR, I think it's just good to consider the [only separate frames workflow](https://master.pfe-preview.zooniverse.org/projects/markb-panoptes/test-project-1-mb-fem-lab/classify?reload=0&workflow=3429&env=staging) and if it'll be supported in FEM (ok if not supported, we'll just note accordingly).

**Comment by goplayoutside3** [comment]
Thanks! That makes more sense, and I think I'll go ahead with merging this PR, building the flipbook view for now, and then check-in with the team on Slack to see if we want to build a totally different separate frames viewer in FEM.

**Pull request merged**